### PR TITLE
[WIP] load an excerpt of a csv into datastore

### PR DESCRIPTION
**Pull request by stefina**
This is my very rookie and WIP approach to make it possible to load only a part of a csv into the datastore if the file is bigger than the configured `max_content_length`

It will try to upload the first few configured number of rows.

It is lacking tests as this is so far a POC to discuss if this is beneficial and how this should be configured.

### CODE REVIEW

File: ckanext/xloader/jobs.py [hunk 1]
```diff
@@ -190,25 +192,40 @@ def get_url():
...
+            if not UPLOAD_EXCERPT:
+                error_msg = 'Resource too large to download: ' \
+                    '{cl} > max ({max_cl}).' \
+                    .format(cl=cl, max_cl=MAX_CONTENT_LENGTH)
+                logger.warning(error_msg)
+                logger.warning('Consider setting option xloader-option '
+                               'ckanext.xloader.upload_excerpt to True to try '
+                               'loading bigger files partially to DataStore.')
+                raise JobError(error_msg)
+            else:
```
Comment (metaodi) [inline ckanext/xloader/jobs.py]: this else is unnecessary, remove it and un-indent the block

File: ckanext/xloader/jobs.py [hunk 2]
```diff
@@ -190,25 +192,40 @@ def get_url():
...
-            tmp_file.write(chunk)
-            m.update(chunk)
+            if not UPLOAD_EXCERPT:
+                error_msg = 'Resource too large to download: ' \
+                    '{cl} > max ({max_cl}).' \
+                    .format(cl=cl, max_cl=MAX_CONTENT_LENGTH)
+                logger.warning(error_msg)
+                logger.warning('Consider setting option xloader-option '
+                               'ckanext.xloader.upload_excerpt to True to try '
+                               'loading bigger files partially to DataStore.')
```
Comment (metaodi) [inline ckanext/xloader/jobs.py]: I personally would remove this warning message. It clutters the log with not very useful information.

File: ckanext/xloader/jobs.py [hunk 3]
```diff
@@ -190,25 +192,40 @@ def get_url():
...
+                if UPLOAD_EXCERPT:
+                    r = requests.get(url, stream=True)
+                    if r.encoding is None:
+                        r.encoding = 'utf-8'
+                    for i, line in enumerate(r.iter_lines()):
+                        if i < MIN_ROW_NUMBER + 1 and line:
+                            tmp_file.write(line + '\n')
+                    length = tmp_file.tell()
+                    m = hashlib.md5()
+                    m.update(tmp_file.read())
```
Comment (metaodi) [inline ckanext/xloader/jobs.py]: move this whole block further down, when you really know if the file is too big.

**Review by metaodi — CHANGES_REQUESTED** [review]
As discuss consider to add a new `FileTooBigError` and catch this error to try adding the excerpt. If the option is not set or the excepert generation fails, you can then raise a `JobError` from your new code.

**Comment by stefina** [comment]
Closing this as this is a duplicate to https://github.com/ckan/ckanext-xloader/pull/63

**Pull request closed without merging**